spec/certificate_authority_spec.rb

```diff
@@ -28,6 +28,58 @@
     end
   end
 
+  context "#http" do
+    before do
+      subject.http = true
+    end
+
+    it "without ipa client should not install" do
+      ipa_configured(false)
+      expect { subject.activate }.to raise_error(ArgumentError, /ipa client/)
+    end
+
+    it "should configure http" do
+      ipa_configured(true)
+      expect_run(/getcert/, anything, response) # getcert returns: the certificate already exist
+
+      expect(LinuxAdmin::Service).to receive(:new).and_return(double("Service", :restart => true))
+      expect(LinuxAdmin::Service).to receive(:new).and_return(double(:enable => double(:start => nil)))
+      expect(FileUtils).to receive(:chmod).with(0o644, anything)
+
+      expect(subject).to receive(:say).twice
+      subject.activate
+      expect(subject.http).to eq(:complete)
+      expect(subject.status_string).to eq("http: complete")
+      expect(subject).to be_complete
+    end
+  end
+
+  context "#postgres client" do
+    before do
+      subject.pgclient = true
+    end
+
+    it "without ipa client should not install" do
+      ipa_configured(false)
+      expect { subject.activate }.to raise_error(ArgumentError, /ipa client/)
+    end
+
+    it "should configure postgres client" do
+      ipa_configured(true)
```

Comment:
Didn't we have a conversation that ended with us deciding that IPA had nothing to do with PG certs? If that's the case, I don't understand this logic.